src/dlgTriggerEditor.cpp

```diff
@@ -3968,12 +3969,11 @@ void dlgTriggerEditor::saveTrigger()
     for (int i = 0; i < 50; i++) {
         QString pattern = mTriggerPatternEdit.at(i)->lineEdit_pattern->text();
         int patternType = mTriggerPatternEdit.at(i)->comboBox_patternType->currentIndex();
-        if (pattern.isEmpty() && patternType != REGEX_PROMPT) {
+        if (pattern.isEmpty() && patternType != REGEX_PROMPT && patternType != REGEX_LINE_SPACER) {
             continue;
         }
-        regexList << pattern;
 
-        switch (mTriggerPatternEdit.at(i)->comboBox_patternType->currentIndex()) {
+        switch (patternType) {
```

Comment:
Since you are using `patternType` directly here, it would make sense to use the same defines in the `case XXX:`s.

Reply:
Forgive me my lack of knowledge, but not sure what are you talking here about. I am actually not sure why this switch was there instead of just appending index directly to list.

Reply:
🤔 Ah, probably for code robustness - so that the index was not hard coded to the values defined in the `REGEX_XXXX` values (bit of a bad choice - TRIGGERTYPE_XXXX` would have been a better one IMVHO).

Yeah, in hindsight my suggestion isn't a change that should be made after all. I.e. ignore my previous comment... 🤦‍♂️